live-objects/src/main/kotlin/io/ably/lib/objects/ObjectMessage.kt

```diff
@@ -0,0 +1,317 @@
+   * excluding the initial value from the create operation if it is a Map object type.
+   * Spec: OST2e
+   */
+  val map: ObjectMap? = null,
+
+  /**
+   * The data that represents the result of applying all operations to a Counter object
+   * excluding the initial value from the create operation if it is a Counter object type.
+   * Spec: OST2f
+   */
+  val counter: ObjectCounter? = null
+)
+
+/**
+ * An @ObjectMessage@ represents an individual object message to be sent or received via the Ably Realtime service.
+ * Spec: OM1
+ */
+internal data class ObjectMessage(
+  /**
+   * unique ID for this object message. This attribute is always populated for object messages received over REST.
+   * For object messages received over Realtime, if the object message does not contain an @id@,
+   * it should be set to @protocolMsgId:index@, where @protocolMsgId@ is the id of the @ProtocolMessage@ encapsulating it,
+   * and @index@ is the index of the object message inside the @state@ array of the @ProtocolMessage@
+   * Spec: OM2a
+   */
+  val id: String? = null,
+
+  /**
+   * time in milliseconds since epoch. If an object message received from Ably does not contain a @timestamp@,
+   * it should be set to the @timestamp@ of the encapsulating @ProtocolMessage@
+   * Spec: OM2e
+   */
+  val timestamp: Long? = null,
+
+  /**
+   * Spec: OM2b
+   */
+  val clientId: String? = null,
+
+  /**
+   * If an object message received from Ably does not contain a @connectionId@,
+   * it should be set to the @connectionId@ of the encapsulating @ProtocolMessage@
+   * Spec: OM2c
+   */
+  val connectionId: String? = null,
+
+  /**
+   * JSON-encodable object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types,
+   * JSON-encodable objects or JSON-encodable arrays. The @extras@ field is provided to contain message metadata and/or
+   * ancillary payloads in support of specific functionality. For 3.1 no specific functionality is specified for
+   * @extras@ in object messages. Unless otherwise specified, the client library should not attempt to do any filtering
+   * or validation of the @extras@ field itself, but should treat it opaquely, encoding it and passing it to realtime unaltered
+   * Spec: OM2d
+   */
+  val extras: Any? = null,
+
+  /**
+   * Describes an operation to be applied to an object.
+   * Mutually exclusive with the `object` field. This field is only set on object messages if the `action` field of the
+   * `ProtocolMessage` encapsulating it is `OBJECT`.
+   * Spec: OM2f
+   */
+  val operation: ObjectOperation? = null,
+
+  /**
+   * Describes the instantaneous state of an object.
+   * Mutually exclusive with the `operation` field. This field is only set on object messages if the `action` field of
+   * the `ProtocolMessage` encapsulating it is `OBJECT_SYNC`.
+   * Spec: OM2g
+   */
+  val `object`: ObjectState? = null,
```

Comment:
Added spec comment for this field -> https://github.com/ably/specification/pull/298/files#r2108964713